File: src/main/java/metaside/Beast.java
package metaside;

import java.util.Random;

/**
 * Created by Petr on 15.10.2016.
 */
public abstract class Beast {
    private int power;

    public Beast(int minPower, int maxPower){
        Random rand = new Random(System.currentTimeMillis());
        this.power = BattleRandom.getInstance().intBetween(minPower, maxPower);
    }

    public int getPower() {
        return power;
    }
}

File: CSharpVersion/MiddleEarth/MiddleEarth/metaside/Beast.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MiddleEarth.metaside
{
    class Beast
    {
        private int power;

        public Beast(int minPower, int maxPower)
        {
            Random rand = new Random(Guid.NewGuid().GetHashCode());
            this.power = rand.Next(minPower, maxPower);
        }

        public int getPower()
        {
            return power;
        }
    }
}
